File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    var answer = 0;
    const N= numbers.length
    function dfs(idx, total, lst){
        if(idx==N){
            // console.log(total, lst)
            if(total==target){
                answer++
            }
            return
        }
        dfs(idx+1, total+numbers[idx]*1, [...lst,numbers[idx]*1 ])
        dfs(idx+1, total+numbers[idx]*(-1), [...lst,numbers[idx]*(-1) ])
    }
    dfs(0,0,[])
    return answer;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = []
    N = len(numbers)
    
    def dfs(idx, total):
        if(idx ==N):
            if(total==target):
                answer.append(1)
            return
        dfs(idx+1, total+numbers[idx])
        dfs(idx+1, total+numbers[idx]*(-1))
    dfs(0,0)
    return len(answer)